homeassistant/components/wled/coordinator.py

```diff
@@ -33,9 +38,69 @@ def update_listeners(self) -> None:
         for update_callback in self._listeners:
             update_callback()
 
+    async def _use_websocket(self) -> None:
+        """Use WebSocket for updates, instead of polling."""
+        if self.wled.connected:
+            # We are already connected
+            return
+
+        # Connect to WebSocket
+        try:
+            await self.wled.connect()
+        except WLEDError as err:
+            self.logger.info(err)
+            return
+
+        async def close_websocket(_) -> None:
+            """Close WebSocket connection."""
+            await self.wled.disconnect()
+
+        # Clean disconnect WebSocket on Home Assistant shutdown
+        self.unsub = self.hass.bus.async_listen_once(
+            EVENT_HOMEASSISTANT_STOP, close_websocket
+        )
+
+        def update_data_from_websocket(device: WLEDDevice) -> None:
+            """Call when WLED reports a state change."""
+            self.data = device
+            self.update_listeners()
```

Comment:
The update coordinator actually has a helper for this: `self.async_set_updated_data`. It will already reset the polling too. 

https://github.com/home-assistant/core/blob/dev/homeassistant/helpers/update_coordinator.py#L268

Reply:
Thanks, that cleans up nicely :)